qubesmanager/qube_manager.py

```diff
@@ -1249,13 +1249,33 @@ def action_shutdownvm_triggered(self):
     def shutdown_vm(self, vm, shutdown_time=vm_shutdown_timeout,
                     check_time=vm_restart_check_timeout, and_restart=False):
         try:
-            vm.shutdown()
+            connected_vms = [x for x in vm.connected_vms if x.is_running()]
+
+            if len(connected_vms) > 0:
+                reply = QMessageBox.question(
+                    self, self.tr("Qube Shutdown Confirmation"),
+                    self.tr("There are some qubes connected to <b>'{0}'</b>!"
+                        "<br><small>Do you want to shutdown: </small>"
+                        "<b>'{1}'</b>?").format(vm.name,
+                            ", ".join([x.name for x in connected_vms])),
+                    QMessageBox.Yes | QMessageBox.Cancel)
```

Comment:
If you for example try to shutdown sys-net, it will first ask this question with sys-firewall as a connected VM, and if you confirm, it will ask again, with other VMs this time. I think it should rather list all of them in the first confirmation.
On the other hand, if you want to just restart a network-providing VM (like sys-net or sys-firewall), it should work in most cases without the need to shutdown connected VMs (but you need to pass force=True).

Reply:
Should we ask for additional confirmation if the VM has connected VMs? (in restart case)

Reply:
Since restarting is not very dangerous (for the connected vms) I think that asking again is not necessary.

Reply:
@marmarek
> On the other hand, if you want to just restart a network-providing VM (like sys-net or sys-firewall), it should work in most cases without the need to shutdown connected VMs (but you need to pass force=True).

One Windows HVM crashed after restarting its network VM. Maybe we should test a little this before merging. I will check with Linux HVM's and version 4.1 instead 4.0.

Reply:
I tested with two Linux based (Alpinelinux and Fedora) and both worked perfect when rebooting sys-firewall. I can't check with Windows guest since I can't boot a restored Windows VM.

Reply:
Well, at least it doesn't crash but network doesn't work with Windows 7.  Maybe should it show a Warning and ask for confirmation? Do you think it could be fixed?